src/main/java/igrad/storage/JsonAdaptedRequirement.java

```diff
@@ -0,0 +1,73 @@
+package igrad.storage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import igrad.commons.exceptions.IllegalValueException;
+import igrad.model.module.Module;
+import igrad.model.module.ModuleCode;
+import igrad.model.requirement.Requirement;
+import igrad.model.requirement.Title;
+
+/**
+ * Jackson-friendly version of {@link Requirement}.
+ */
+class JsonAdaptedRequirement {
```

Comment:
nice you managed to get the storage working, I'll reference this code then, to get my CourseInfo working too 👍